src/lib/plot/axis/x-axis.js

```diff
@@ -31,14 +32,21 @@ const propTypes = {
   ])
 };
 
+delete propTypes.axisType;
```

Comment:
Using `delete` prevents optimizations and it's a mutating operation. How about alternatives such as the rest operator or using lodash's `omit`?